kotlin/workflow-core/src/main/java/com/squareup/workflow/WorkflowAction.kt

```diff
@@ -83,70 +110,85 @@ interface WorkflowAction<StateT, out OutputT : Any> {
      * Convenience function that returns a [WorkflowAction] that will just set the state to [newState]
      * (without considering the current state) and optionally emit an output.
      */
-    @Deprecated("Implement WorkflowAction directly, or use `WorkflowAction.invoke`")
+    @Deprecated(
+        "Implement WorkflowAction directly, " +
+            "or use StatefulWorkflow.action or StatelessWorkflow.action"
+    )
     fun <StateT, OutputT : Any> enterState(
       newState: StateT,
       emittingOutput: OutputT? = null
     ): WorkflowAction<StateT, OutputT> =
       WorkflowAction({ "enterState($newState, $emittingOutput)" }) {
-        state = newState
-        return@WorkflowAction emittingOutput
+        nextState = newState
+        emittingOutput?.let { nextOutput = it }
       }
 
     /**
      * Convenience function that returns a [WorkflowAction] that will just set the state to [newState]
      * (without considering the current state) and optionally emit an output.
      */
-    @Deprecated("Implement WorkflowAction directly, or use `WorkflowAction.invoke`")
+    @Deprecated(
+        "Implement WorkflowAction directly, " +
+            "or use StatefulWorkflow.action or StatelessWorkflow.action"
+    )
     fun <StateT, OutputT : Any> enterState(
       name: String,
       newState: StateT,
       emittingOutput: OutputT? = null
     ): WorkflowAction<StateT, OutputT> =
       WorkflowAction({ "enterState($name, $newState, $emittingOutput)" }) {
-        state = newState
-        return@WorkflowAction emittingOutput
+        nextState = newState
+        emittingOutput?.let { nextOutput = it }
       }
 
     /**
      * Convenience function to implement [WorkflowAction] without returning the output.
      */
-    @Deprecated("Implement WorkflowAction directly, or use `WorkflowAction.invoke`")
+    @Deprecated(
+        "Implement WorkflowAction directly, " +
+            "or use StatefulWorkflow.action or StatelessWorkflow.action"
+    )
     fun <StateT, OutputT : Any> modifyState(
       name: () -> String,
       emittingOutput: OutputT? = null,
       modify: (StateT) -> StateT
     ): WorkflowAction<StateT, OutputT> =
       WorkflowAction({ "modifyState(${name()}, $emittingOutput)" }) {
-        state = modify(state)
-        return@WorkflowAction emittingOutput
+        nextState = modify(nextState)
+        emittingOutput?.let { nextOutput = it }
       }
 
     /**
      * Convenience function to implement [WorkflowAction] without changing the state.
      */
-    @Deprecated("Use `WorkflowAction { output }`")
+    @Deprecated(
+        "Implement WorkflowAction directly, " +
+            "or use StatefulWorkflow.action or StatelessWorkflow.action"
+    )
```

Comment:
Adding replacement strategy here and for other deprecated APIs would make migration easier:
```kotlin
@Deprecated(
    message = "Implement WorkflowAction directly, " +
        "or use StatefulWorkflow.action or StatelessWorkflow.action",
    replaceWith = ReplaceWith(
        expression = "action { nextOutput = output }",
        imports = arrayOf("com.squareup.workflow.action")
    )
)
```

Reply:
I've added them, but as usual I'm finding that they don't actually work. 

I suppose they at least get the import in place, and provide a bit more documentation.

Reply:
In my experience replacements do work in IDE but imports are not always added for whatever reason. 😕